components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -54,124 +67,527 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
+                                     attribute:NSLayoutAttributeWidth
+                                     relatedBy:NSLayoutRelationEqual
+                                        toItem:nil
+                                     attribute:NSLayoutAttributeNotAnAttribute
+                                    multiplier:1
+                                      constant:kIconImageViewSideLength];
+
+    NSLayoutConstraint *iconImageHeightConstraint =
+        [NSLayoutConstraint constraintWithItem:iconImageView
+                                     attribute:NSLayoutAttributeHeight
+                                     relatedBy:NSLayoutRelationEqual
+                                        toItem:nil
+                                     attribute:NSLayoutAttributeNotAnAttribute
+                                    multiplier:1
+                                      constant:kIconImageViewSideLength];
+
+    [NSLayoutConstraint
+        activateConstraints:@[ iconImageWidthConstraint, iconImageHeightConstraint ]];
+
+    iconImageView.contentMode = UIViewContentModeCenter;
+    iconImageView.clipsToBounds = YES;
+    self.iconImageView = iconImageView;
+    [self.containerView addSubview:self.iconImageView];
+    self.isLayoutDirty = YES;
   } else {
-    _image = nil;
-    [self.iconImageViewContainer removeFromSuperview];
-    self.iconImageViewContainer = nil;
+    self.iconImageView.image = icon;
   }
   [self setNeedsLayout];
 }
 
-- (void)setNumberOfButtons:(NSUInteger)numberOfButtons {
-  NSAssert(numberOfButtons <= kNumberOfButtonsLimit,
-           @"%@ class doesn't support more than %lu buttons", NSStringFromClass([self class]),
-           (unsigned long)kNumberOfButtonsLimit);
-  if (numberOfButtons != self.buttons.count) {
-    NSMutableArray *mutableButtons = [self.buttons mutableCopy];
-    if (numberOfButtons > self.buttons.count) {
-      for (NSUInteger index = self.buttons.count; index < numberOfButtons; ++index) {
-        MDCButton *button = [[MDCButton alloc] init];
-        [mutableButtons addObject:button];
-        [self addSubview:button];
-      }
-    } else if (numberOfButtons < self.buttons.count) {
-      for (NSUInteger index = numberOfButtons; index < self.buttons.count; ++index) {
-        MDCButton *lastObject = [mutableButtons lastObject];
-        [lastObject removeFromSuperview];
-        [mutableButtons removeLastObject];
-      }
-    }
-    self.buttons = [mutableButtons copy];
+- (UIImage *)icon {
+  return self.iconImageView.image;
+}
+
+- (void)setButtons:(NSMutableArray<MDCButton *> *_Nonnull)buttons {
+  NSAssert(buttons.count <= kNumberOfButtonsLimit, @"%@ doesn't support more than %lu buttons",
+           NSStringFromClass([self class]), (unsigned long)kNumberOfButtonsLimit);
+  _buttons = buttons;
+  for (MDCButton *button in buttons) {
+    button.translatesAutoresizingMaskIntoConstraints = NO;
+    [self.containerView addSubview:button];
   }
+  self.isLayoutDirty = YES;
```

Comment:
I believe we can use `[self invalidateIntrinsicContentSize]`

Reply:
Same reason as mentioned above in https://github.com/material-components/material-components-ios/pull/6830#discussion_r264361551. Marking as resolved to keep the conversation in a single thread.